gcamwrapper/query_library.py

```diff
@@ -0,0 +1,251 @@
+import yaml
+import warnings
+import re
+import os.path as path
+import pkg_resources
+
+def read_yaml(yaml_file):
+    '''Read a YAML file.
+       
+    :param yaml_file: The file path to the YAML file to load
+    :type yaml_file: str
+
+    :returns: A dict representing the YAML data
+    '''
+
+    with open(yaml_file, 'r') as yml:
+        return yaml.load(yml, Loader=yaml.FullLoader)
+
+'''The parsed query file included with the package'''
+PACKAGE_QUERIES = yaml.load(pkg_resources.resource_stream(__name__, 'query_library.yml'), Loader=yaml.FullLoader)
+
+class Query(str):
+    '''A Simple extension to str to be able to associate units meta data'''
+    units: None
+
+def get_query(*args, query_file = None):
+    '''Look up a query from a YAML file.
+       The YAML query files will often be organized into nested categories
+       so users must provide the full path as individual strings such as:
+       \code{get_query("emissions", "co2_emissions")}
+
+    :param query_file: The path to a YAML file to read if not None otherwise
+                       the package query file will be used.
+    :type query_file: str
+    :param *args: Strings specifiying the path to the query to lookup
+    :type *args: str
+
+    :returns: The raw query from the query file with units set as an attribute
+              if specified
+    '''
+
+    if query_file is None:
+        queries = PACKAGE_QUERIES
+    else:
+        queries = read_yaml(query_file)
+    query = queries
+    for path in args:
+        try:
+            query = query[path]
+        except KeyError:
+            raise Exception("Could not find query: ", args)
```

Comment:
No need to state `raise Exception...` this will mask the real `KeyError` so you can just do:

```python
raise(f"Could not find query:  {args}")
```